migrations/20231005230721_dynamic-fields.sql

```diff
@@ -0,0 +1,145 @@
+CREATE TABLE games (
+	id int PRIMARY KEY, -- Only used in db
+	name varchar(64),
+  CONSTRAINT unique_game_name UNIQUE (name)
+);
+INSERT INTO games(id, name) VALUES (1, 'minecraft-java');
+INSERT INTO games(id, name) VALUES (2, 'minecraft-bedrock');
+
+-- we are creating a new loader type- 'mrpack'- for minecraft modpacks
+INSERT INTO loaders (loader) VALUES ('mrpack');
+INSERT INTO loaders_project_types (joining_loader_id, joining_project_type_id) SELECT DISTINCT l.id, pt.id FROM loaders l CROSS JOIN project_types pt WHERE pt.name = 'modpack' AND l.loader = 'mrpack';
+
+-- We create 'modpack' categories for every loader 
+-- That way we keep information like "this modpack is a fabric modpack"
+INSERT INTO categories (category, project_type) 
+SELECT DISTINCT l.loader, pt.id FROM loaders l CROSS JOIN project_types pt WHERE pt.name = 'modpack' AND l.loader != 'mrpack';
+
+-- insert the loader of every modpack mod as a category
+INSERT INTO mods_categories (joining_mod_id, joining_category_id)
+SELECT DISTINCT m.id, c.id 
+FROM mods m
+LEFT JOIN versions v ON m.id = v.mod_id
+LEFT JOIN loaders_versions lv ON v.id = lv.version_id
+LEFT JOIN loaders l ON lv.loader_id = l.id
+CROSS JOIN categories c 
+WHERE m.project_type = (SELECT id FROM project_types WHERE name = 'modpack') AND c.category = l.loader;
+
+-- Non mrpack loaders no longer support modpacks
+DELETE FROM loaders_project_types WHERE joining_loader_id != (SELECT id FROM loaders WHERE loader = 'mrpack') AND joining_project_type_id = (SELECT id FROM project_types WHERE name = 'modpack');
```

Comment:
How come the loaders were made into categories? shouldn't they just be per-version dynamic fields?